src/runtime_src/core/common/CMakeLists.txt

```diff
@@ -38,10 +38,14 @@ file(GLOB XRT_CORE_COMMON_LIB_FILES
   "api/*.cpp"
   )
 
+if (${XRT_NATIVE_BUILD} STREQUAL "yes")
+  add_definitions (-DXRT_ENABLE_AIE -DXRT_AIE_BUILD)
+endif()
+
```

Comment:
I think this is the reason for the runtime failures observed in the pipeline.  You should run and debug a few examples to understand the implications on enabling these switches on x86.   For example load_xclbin takes a very different path.

Reply:
Enabling the definitions under core/common, I am under impression that it even for the core/pcie even. If thats not the case, we need to explicitly enable it inside core/pcie even. And you suggested to use src/CMakeLists.txt to add these definitions, wondering by then we have the XRT_NATIVE_BUILD is set appropriately.

Reply:
resulting CMake errors when used the add_compile_definitions, I dont know if this is supported with the CMake version I have.